docs/misc/arm/scf.txt

```diff
@@ -0,0 +1,233 @@
+Share a coprocessor to several domains using guest's Device Tree
+================================================================
+
+This example will use a dummy coprocessosr configuration and is based on an RCar
+Gen3 device tree already adjusted for dom0. Some free iomem range not assigned
+to any real device will be used for a dummy coprocessors definition. One dummy
+coprocessor would have a single iomem range and one irq. Another one will have
+several named iomem ranges and irqs.
+Two virtual instances of each coprocessor will be configured for Dom0 with using
+a system device tree and other two will be configured for domU using a partial
+device tree.
+
+Let us have two dummy coprocessors defined as following:
+
+\ {
+	...
+	aliases {
+		...
+		pcoproc0 = &coproc0;
+		pcoproc1 = &coproc1;
+		...
+	};
+	...
+	soc {
+		...
+		coproc0: pcoproc0@0xe6110000 {
+			compatible = "vendor_xxx,coproc_xxx";
+			reg = <0x0 0xe6110000 0x0 0x1000>;
+			interrupts = <GIC_SPI 160 IRQ_TYPE_LEVEL_HIGH>;
+			status = "okay";
+		};
+
+		coproc1: pcoproc1@0xe6182000 {
+			compatible = "vendor_xxx,coproc_xxc";
+			reg = <0x0 0xe6182000 0x0 0x1000>,
+			      <0x0 0xe6184000 0x0 0x1000>,
+			      <0x0 0xe6188000 0x0 0x8000>;
+			reg-names = "op", "mmu", "sram";
+			interrupts = <GIC_SPI 132 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 135 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "op", "mmu";
+			status = "okay";
+			xen,coproc;
+		};
+
+		...
+	};
+	...
+};
+
+Notes:
+    * If a coprocessor has several iomem ranges they must be named. Same for
```

Comment:
Because on having several ranges/irqs we need to be specific what range/irq belongs to what function. Implementing naming matching we do make a configuration obvious and unambiguous.

In XEN ARM, GIC probe does match ranges by their position in the array. I do not really like that approach.